Orange/widgets/utils/itemmodels.py

```diff
@@ -910,6 +901,25 @@ def row_instance(index):
         if self.__rowCount > (2 ** 31 - 1):
             raise ValueError("len(sourcedata) > 2 ** 31 - 1")
 
+    def _get_source_item(self, row, col):
+        if col < self.Y_len:
+            if self.source.Y.ndim == 1:
+                return self.source.Y[row]
+            elif self.Y_density == Storage.DENSE:
+                return self.source.Y[row, col]
+            else:
+                return self.source.Y[row]
+        elif self.Y_len <= col < self.Y_len + self.M_len:
+            if self.M_density == Storage.DENSE:
+                return self.source.metas[row, col - self.Y_len]
+            else:
+                return self.source.metas[row]
+        else:
+            if self.X_density == Storage.DENSE:
+                return self.source.X[row, col - self.Y_len - self.M_len]
+            else:
+                return self.source.X[row]
```

Comment:
AFAIK accessing X|Y|metas in case of an SQLTable will download a limited number (10000) rows into memory on first access.  Indexing the arrays after that with row indices larger then the download limit (10000+) will raise an IndexError. @lanzagar is SQL still even supported?